components/shared/posts/CreatePost.tsx

```diff
@@ -0,0 +1,141 @@
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    getValues,
+    watch,
+  } = useForm<z.infer<typeof CreatePostSchema>>({
+    resolver: zodResolver(CreatePostSchema),
+    defaultValues: {
+      title: "",
+      postType: "knowledge",
+      description: "",
+      content: "",
+      tags: [],
+      code: "",
+      experiences: [],
+      resourceLinks: [],
+    },
+  });
+
+  const {
+    fields: experience,
+    append: appendExperience,
+    remove: removeExperience,
+  } = useFieldArray<z.infer<typeof CreatePostSchema>, "experiences">({
+    control,
+    name: "experiences",
+  });
+  const {
+    fields: resourceLinks,
+    append: appendResourceLink,
+    remove: removeResourceLink,
+  } = useFieldArray<z.infer<typeof CreatePostSchema>, "resourceLinks">({
+    control,
+    name: "resourceLinks",
+  });
+  const watchContent = watch("content");
+  const onSubmit: SubmitHandler<z.infer<typeof CreatePostSchema>> = async (
+    data
+  ) => {
+    const {
+      title,
+      postType,
+      content,
+      description,
+      tags = [],
+      code,
+      experiences,
+      resourceLinks,
+    } = data;
+    const dbResources =
+      resourceLinks?.map((link) => ({
+        title: link?.title,
+        url: link?.url,
+      })) || [];
+    console.log(data);
+    try {
+      startTransition(async () => {
+        const result = await createNewPost({
+          title,
+          postType,
+          content,
+          tags,
+          code,
+          description,
+          experiences: experiences?.map((experience) => experience.name) || [],
+          resourceLinks: dbResources.map((link) => ({
+            title: link?.title || "",
+            url: link?.url || "",
+          })),
+          author: user?._id!,
+        });
+        if (result) {
+          toast({ title: "Post Created Successfully" });
+        }
+      });
+    } catch (error) {}
+  };
+  useEffect(() => {
+    console.log(watchContent);
+  }, [watchContent]);
+  return (
+    <section>
+      <form
+        onSubmit={handleSubmit(onSubmit)}
+        className="py-7.5 flex w-full flex-1 flex-col gap-6"
+      >
+        <span className="display-1-bold pt-10 text-white-100">
+          Create a Post
+        </span>
+        <NewPostInfo register={register} control={control} errors={errors} />
+        <NewExperience
+          experienceFields={experience}
+          appendExperience={appendExperience}
+          removeExperience={removeExperience}
+          register={register}
+        />
+        <NewContent register={register} control={control} />
+        <NewResourceLink
+          resourceLinks={resourceLinks}
+          appendResourceLink={appendResourceLink}
+          removeResourceLink={removeResourceLink}
+          register={register}
+        />
+
+        <CustomButton buttonType="primary" type="submit" disabled={pending}>
+          Create Post {pending && <LoadingSpinner />}
+        </CustomButton>
```

Comment:
Remember we want to display error messages for each of these!

Right now if I try to submit empty values as a user I don't get any feedback

Reply:
yep im taking an inventory of what i need to do to polish each component - including error feedback, creating and hiding the code block component, adding tags, etc